core/server/master/src/main/java/alluxio/master/meta/UpdateCheck.java

```diff
@@ -98,7 +111,46 @@ public static String getUserAgentString(String clusterID) throws IOException {
         sysInfo = joiner.join(sysInfo, joiner.join(ec2Info));
       }
     }
-    return String.format("Alluxio/%s (%s)", ProjectConstants.VERSION, sysInfo);
+    return sysInfo;
+  }
+
+  /**
+   * Get the features information.
+   *
+   * @return a list of string represent feature names
+   */
+  @VisibleForTesting
+  public static String getFeatureString() {
+    Joiner joiner = Joiner.on("; ").skipNulls();
+    List<String> featureInfo = new ArrayList<>();
+    if (FeatureUtils.isEmbeddedJournal()) {
```

Comment:
this list looks really long and repetitive. not sure if this is possible in java, but could this be represented as a map of string to function, with a loop iterating over the map?